### PR TITLE
Document liveness- and readiness probes

### PR DESCRIPTION
**Pull request by rmohr**

### CODE REVIEW

File: workloads/virtual-machines/probes.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+Readiness Probes are an indicator for Services and Endpoints if the
+VirtualMachineInstance is ready to receive traffic from Services. If Readiness
+Probes fail, the VirtualMachineInstance will be removed from the Endpoints
+which back services until the probe recovers.
+
+## Define a HTTP Liveness Probe
+
+The following VirtualMachineInstance configures a HTTP Liveness Probe via
+`spec.livenessProbe.httpGet`, wich will query port 1500 of the
```
Comment (ILpinto) [inline workloads/virtual-machines/probes.md]: typo: wich will ...
which will  ...

File: workloads/virtual-machines/probes.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+          - setenforce 0
+          - dnf install -y nmap-ncat
+          - systemd-run --unit=httpserver nc -klp 1500 -e '/usr/bin/echo -e HTTP/1.1 200 OK\\n\\nHello World!'
+    name: cloudinitvolume
+```
+
+## Define a TCP Liveness Probe
+
+The following VirtualMachineInstance configures a TCP Liveness Probe via
+`spec.livenessProbe.tcpSocket`, wich will query port 1500 of the
```
Comment (petrkotas) [inline workloads/virtual-machines/probes.md]: s/wich/which

Comment (petrkotas) [inline workloads/virtual-machines/probes.md]: s/wich/which

**Comment by rmohr** [comment]
Typos addressed. Thanks.

File: workloads/virtual-machines/probes.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+        chpasswd: { expire: False }
+        bootcmd:
+          - setenforce 0
+          - dnf install -y nmap-ncat
+          - systemd-run --unit=httpserver nc -klp 1500 -e '/usr/bin/echo -e HTTP/1.1 200 OK\\n\\nHello World!'
+    name: cloudinitvolume
+```
+
+Note that in the case of Readiness Probes, it is also possible to set a
+`failureThreshold` and a `successThreashold` to only flip between ready- and
```
Comment (stu-gott) [inline workloads/virtual-machines/probes.md]: should `ready-` be `ready` here?

Comment (rmohr) [reply workloads/virtual-machines/probes.md]: Fixed. Thanks.

**Pull request merged**